src/CSET/operators/collapse.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 Met Office and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Operators to perform various kind of collapse on either 1 or 2 dimensions.
+"""
+
+import iris
+import iris.cube
+import iris.analysis
+
+
+def collapse_1dim(
+    cube: iris.cube.Cube, coordinate: str, method: str, **kwargs
+) -> iris.cube.Cube:
+    """
+    Collapses similar (stash) fields in a cube into a cube collapsing around the specified coordinate and method.
+    This could be a (weighted) mean or percentile.
+
+    Arguments
+    ---------
+    cube: iris.cube.Cube
+         Cube to collapse and iterate over one dimension
+    coordinate: str
+         Coordinate to collapse over i.e.
+         'time', 'longitude', 'latitude','model_level_number'
+    method: str
+         Type of collapse i.e. method: 'MEAN', 'MAX', 'MIN', 'MEDIAN', 'PERCENTILE'
+         getattr creates iris.analysis.MEAN, etc
+         For PERCENTILE YAML file requires i.e.
+         method: 'PERCENTILE'
+         additional_percent: 90
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+         Single variable but several methods of aggregation
+
+
+    Raises
+    ------
+    ValueError
+    If the constraint doesn't produce a single cube containing a field.
+    """
+
+    if method != "PERCENTILE":
+        collapsed_cube = cube.collapsed(coordinate, getattr(iris.analysis, method))
+    if method == "PERCENTILE":
+        for num in kwargs.values():
+            collapsed_cube = cube.collapsed(
+                coordinate, getattr(iris.analysis, method), percent=num
+            )
+
+    return collapsed_cube
```

Comment:
```suggestion
    if method == "PERCENTILE":
        if not additional_percent:
            raise ValueError("Must specify additional_percent")
        collapsed_cube = cube.collapsed(
            coordinate, iris.analysis.PERCENTILE, percent=additional_percent
        )
        return collapsed_cube
    collapsed_cube = cube.collapsed(coordinate, getattr(iris.analysis, method))
    return collapsed_cube
```
If you flip the order of the checks around, and return early, you can skip an unnecessary if.
I've also directly used the additional_percent added in the function signature above.